generic3g/MAPL_Generic.F90

```diff
@@ -402,6 +407,62 @@ subroutine gridcomp_add_varspec_basic(gridcomp, variable_spec, rc)
       _RETURN(_SUCCESS)
    end subroutine gridcomp_add_varspec_basic
 
+   subroutine gridcomp_add_fieldspec( &
+        gridcomp, &
+        state_intent, &
+        short_name, &
+        standard_name, &
+        dims, &
+        vstagger, &
+        ! OPTIONAL
+        unusable, &
+        units, &
+        ungridded_dims, &
+        restart, &
+        rc)
+      type(ESMF_GridComp), intent(inout) :: gridcomp
+      type(Esmf_StateIntent_Flag), intent(in) :: state_intent
+      character(*), intent(in) :: short_name
+      character(*), intent(in) :: standard_name
+      character(*), intent(in) :: dims
+      type(VerticalStaggerLoc), intent(in) :: vstagger
+      ! OPTIONAL
+      class(KeywordEnforcer), optional, intent(in) :: unusable
+      character(*), optional, intent(in) :: units
+      integer, optional, intent(in) :: ungridded_dims(:)
+      logical, optional, intent(in) :: restart
+      integer, optional, intent(out) :: rc
+
+      type(ESMF_StateItem_Flag), parameter :: itemtype = ESMF_STATEITEM_FIELD
+      type(VariableSpec) :: var_spec
+      type(HorizontalDimsSpec) :: horizontal_dims_spec
+      type(OuterMetaComponent), pointer :: outer_meta
+      type(ComponentSpec), pointer :: component_spec
+      integer :: status
+
+      _ASSERT((dims=="xyz") .or. (dims=="xy") .or. (dims=="z"), "dims can be one of xyz/xy/z")
+      horizontal_dims_spec = HORIZONTAL_DIMS_GEOM
```

Comment:
There is something interesting about the fact that no special logic is needed for vertical here, but horizontal needs it.   Not immediately seeing anything better, but bringing it to other intention in case they do.

Reply:
Indeed.